src/log_analyzer.py

```diff
@@ -126,6 +128,16 @@ def run_agent_analysis(error_summary, product, product_config):
         reraise=True,
     )
     def _run_agent_analysis():
+        if ANALYSIS_MODE == "gemini":
```

Comment:
I believe regulating inference through ENV is temporary until we decide on a legit forefront. Just making sure that we are inline here.